src/lib/transaction_snark/test/ring_sig.ml

```diff
@@ -266,7 +266,8 @@ let%test_unit "ring-signature snapp tx with 3 parties" =
                 respond Unhandled
           in
           let pi : Pickles.Side_loaded.Proof.t =
-            (fun () -> ringsig_prover ~handler [] tx_statement)
+            (fun () ->
+              ringsig_prover ~handler [] tx_statement |> Promise.to_deferred)
```

Comment:
Can we do something about the types returned by `ringsig_prover` and other rules? Presumably it should be possible to create a `Pickles.compile_promise` with the new type and generate `Pickles.compile` from it for regular use.

Reply:
Argh... I think that means I have to map an Hlist :fearful: :cry: